src/panels/calendar/ha-panel-calendar.js

```diff
@@ -0,0 +1,175 @@
+    return html`
+      <style include="iron-flex ha-style">
+        .content {
+          padding: 16px;
+          @apply --layout-horizontal;
+        }
+
+        ha-big-calendar {
+          min-height: 500px;
+          min-width: 100%;
+        }
+
+        #calendars {
+          padding-right: 16px;
+          width: 15%;
+        }
+
+        div.all_calendars {
+    ￼     height: 20px;
+    ￼     text-align: center;
+        }
+
+        span.calendar_color {
+          width: 15px;
+          height: 5px;
+     ￼    margin: 0px 10px 2px 0px;
+     ￼    display: inline-block;
+          background-color: #3174ad;
+        }
+     
+        span.calendar_color_spacer {
+          width: 15px;
+        }
+
+        .iron-selected {
+          background-color: #e5e5e5;
+          font-weight: normal;
+        }  
+      </style>
+
+    <app-header-layout has-scrolling-region>
+      <app-header slot="header" fixed>
+        <app-toolbar>
+          <ha-menu-button narrow='[[narrow]]' show-menu='[[showMenu]]'></ha-menu-button>
+          <div main-title>[[localize('panel.calendar')]]</div>
+        </app-toolbar>
+      </app-header>
+
+      <div class="flex content">
+        <div id="calendars" class="layout vertical wrap">
+          <paper-card heading="Calendars">
+            <div class="all_calendars">
+              <paper-checkbox id="all_calendars" on-change="checkAll" checked>All calendars</paper-checkbox>
+            </div>
+            <paper-listbox id="calendar_list" multi on-selected-items-changed="_fetchData" selected-values="{{selectedCalendars}}" attr-for-selected="item-name">
+              <template is="dom-repeat" items="[[calendars]]">
+                <paper-item item-name=[[item.name]]>
+                  <span class="calendar_color" style="background-color: [[item.color]]"></span>
+                  <span class="calendar_color_spacer"></span>
+                  [[item.name]]
+                </paper-item>
+              </template>
+            </paper-listbox>
+          </paper-card>
+        </div>
+        <div class="flex layout horizontal wrap">
+          <ha-big-calendar events=[[items]]></ha-big-calendar>
+        </div>
+      </div>
+    </app-header-layout>
+    `;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._fetchData = this._fetchData.bind(this);
+    // TODO implement calendar_updated event
+    // this.hass.connection.subscribeEvents(this._fetchData, 'calendar_updated')
+    //  .then(function (unsub) { this._unsubEvents = unsub; }.bind(this));
+    this._fetchData();
+  }
+
+  _fetchData() {
+    // Fetch data
+    // Fetch calendar list
+    this.hass.callApi('get', 'calendar-list')
+      .then(function (items) {
+        this.calendars = items;
+      }.bind(this));
+    // Fecth events fro selected calendar
+    this.items = [];
+    this.hass.callApi('post', 'calendar', { calendars: this.selectedCalendars })
+      .then(function (items) {
```

Comment:
use arrow functions

Reply:
I don't understand this comment (I'm not really good in javascript) Could you give me more details ?

Reply:
```
.then((items) => {
  this.items = items;
});
```

Reply:
@c727 Thanks ! Is `.bind(this));` still needed ?

Reply:
no problem, you don't need bind(this) anymore